### PR TITLE
Allow adding published to the generated Cargo.toml

### PR DESCRIPTION
**Pull request by hatchan**

### CODE REVIEW

**Review by stephlow — APPROVED** [review]

**Review by gagbo — APPROVED** [review]

File: fp-bindgen/src/generators/mod.rs [hunk 1]
```diff
@@ -68,6 +68,10 @@ pub struct RustPluginConfig {
 
     /// The license of the generated crate.
     pub license: Option<RustPluginConfigValue>,
+
+    /// Whether the crate is marked as published or not (note: this should
+    /// the string value "true" or "false")
+    pub published: Option<RustPluginConfigValue>,
```
Comment (gagbo) [inline fp-bindgen/src/generators/mod.rs:74]:
```suggestion
    pub publish: Option<RustPluginConfigValue>,
```

File: fp-bindgen/src/generators/mod.rs [hunk 2]
```diff
@@ -81,6 +85,7 @@ impl RustPluginConfig {
                 description: None,
                 readme: None,
                 license: None,
+                published: None,
```
Comment (gagbo) [inline fp-bindgen/src/generators/mod.rs:88]:
```suggestion
                publish: None,
```

File: fp-bindgen/src/generators/mod.rs [hunk 3]
```diff
@@ -178,6 +183,11 @@ impl RustPluginConfigBuilder {
         self
     }
 
+    pub fn published(mut self, value: bool) -> Self {
+        self.config.published = Some(RustPluginConfigValue::String(value.to_string()));
+        self
+    }
+
```
Comment (gagbo) [inline fp-bindgen/src/generators/mod.rs:190]:
```suggestion
    pub fn publish(mut self, value: bool) -> Self {
        self.config.publish = Some(RustPluginConfigValue::String(value.to_string()));
        self
    }

```

File: fp-bindgen/src/generators/rust_plugin/mod.rs [hunk 4]
```diff
@@ -112,6 +112,7 @@ fn generate_cargo_file(
             format_cargo_key("description", config.description),
             format_cargo_key("readme", config.readme),
             format_cargo_key("license", config.license),
+            format_cargo_key("published", config.published),
```
Comment (gagbo) [inline fp-bindgen/src/generators/rust_plugin/mod.rs:115]:
```suggestion
            format_cargo_key("publish", config.published),
```

**Review by gagbo — CHANGES_REQUESTED** [review]
Abort mission, [the key is `publish`](https://doc.rust-lang.org/cargo/reference/manifest.html#the-publish-field)

**Comment by hatchan** [comment]
> Abort mission, [the key is `publish`](https://doc.rust-lang.org/cargo/reference/manifest.html#the-publish-field)

Yeah I just noticed it as well :)

**Review by gagbo — APPROVED** [review]

**Pull request merged**